kernel/uint63_63.ml

```diff
@@ -179,6 +200,11 @@ let compare (x:int) (y:int) =
   else if y > x then -1
   else 0
 
+let compares (x : int) (y : int) =
+  if x > y then 1
+  else if y > x then -1
+  else 0
+
```

Comment:
I'm surely missing something, but I recall `y - x` being faster in OCaml that this.

Reply:
I just copy-pasted and adapted the code of `compare`. Maybe both could be changed to `y - x`?

Reply:
> I recall `y - x` being faster in OCaml that this.

Faster, certainly, but correct, definitely not. Consider `max_int` and `min_int` for instance. The latter is smaller, yet their difference is negative due to wrapping.

That said, it might be better to let the compiler find the best way to compile that comparison:
```ocaml
let compares (x : int) (y : int) =
  Int.compare x y
```

(But truth be told, the generated code for `Int.compare` is not that great. Its main advantage is to be branch-free, and thus independent of the branch predictor.)

Reply:
I don't know if the "spec" is to return 1, 0 or -1 exactly here. Usually, any positive, 0, any negative is OK.

As I said, I'm no expert here, but I was looking at the code and this snippet seemed odd to me.

Reply:
```
let compares (x : int) (y : int) =
  Int.compare x y
```
This is now the implementation.